cirq/contrib/quimb/mps_simulator.py

```diff
@@ -263,60 +336,98 @@ def __init__(self, qubit_map, rsum2_cutoff, initial_state=0):
             initial_state = initial_state // d
         self.M = self.M[::-1]
         self.rsum2_cutoff = rsum2_cutoff
+        self.sum_prob_atol = sum_prob_atol
         self.num_svd_splits = 0
 
-    def i_str(self, i):
+    def i_str(self, i: int) -> str:
+        # Returns the indice name for the i'th qid.
         return self.format_i.format(i)
 
-    def mu_str(self, i, j):
-        # By convention, the lower index is always the first.
+    def mu_str(self, i: int, j: int) -> str:
+        # Returns the indice name for the pair of the i'th and j'th qids. Note
+        # that by convention, the lower index is always the first in the output
+        # string.
         smallest = min(i, j)
         largest = max(i, j)
         return self.format_mu.format(smallest, largest)
 
     def __str__(self) -> str:
-        return str([x.data for x in self.M])
+        return str(qtn.TensorNetwork(self.M))
 
     def _value_equality_values_(self) -> Any:
-        return self.qubit_map, self.M, self.rsum2_cutoff
+        return self.qubit_map, self.M, self.rsum2_cutoff, self.sum_prob_atol
 
     def copy(self) -> 'MPSState':
-        state = MPSState(self.qubit_map, self.rsum2_cutoff)
+        state = MPSState(self.qubit_map, self.rsum2_cutoff, self.sum_prob_atol)
         state.M = [x.copy() for x in self.M]
         state.num_svd_splits = self.num_svd_splits
         return state
 
-    def _sum_up(self, skip_tracing_out_for_qubits):
-        end_bras = []
-        if skip_tracing_out_for_qubits:
-            for qubit, i in self.qubit_map.items():
-                if i not in skip_tracing_out_for_qubits:
-                    d = qubit.dimension
-                    data = [math.sqrt(d)] * d
-                    end_bras.append(qtn.Tensor(data=data, inds=(self.i_str(i),)))
-        tensor_network = qtn.TensorNetwork(self.M + end_bras)
-        return tensor_network.contract(inplace=False)
-
-    def partial_state_vector(self, skip_tracing_out_for_qubits):
-        M = self._sum_up(
-            skip_tracing_out_for_qubits={
-                self.qubit_map[qubit] for qubit in skip_tracing_out_for_qubits
-            }
-        )
+    def state_vector(self) -> np.ndarray:
+        """Returns the full state vector.
+
+        Returns:
+            A vector that contains the full state.
+        """
+        tensor_network = qtn.TensorNetwork(self.M)
+        state_vector = tensor_network.contract(inplace=False)
+
         # Here, we rely on the formatting of the indices, and the fact that we have enough
         # leading zeros so that 003 comes before 100.
-        sorted_ind = tuple(sorted(M.inds))
-        return M.fuse({'i': sorted_ind}).data
+        sorted_ind = tuple(sorted(state_vector.inds))
+        return state_vector.fuse({'i': sorted_ind}).data
+
+    def partial_trace(self, keep_qubits: Set[ops.Qid]) -> np.ndarray:
+        """Returns the partial trace only for the keep_qubits specified.
```

Comment:
"Trace out all qubits except `keep_qubits`"

(as written, it's confusing whether keep_qubits are traced out or kept)

Reply:
Done.